radis/io/exomol.py

```diff
@@ -8,14 +8,145 @@
 import pathlib
 
 import numpy as np
+import pandas as pd
 
 from radis.api.exomolapi import (
     MdbExomol,
     get_exomol_database_list,
     get_exomol_full_isotope_name,
+    read_states,
 )
 from radis.db.classes import get_molecule_identifier
 
+# ExoMol state/transition processing and degeneracy calculation logic
+def _map_states_to_transitions(df, states_df):
+    """Map electronic state information from energy levels to transitions."""
+    if 'Es' in states_df.columns:
+        state_map = dict(zip(states_df['i'], states_df['Es']))
+    elif 'state' in states_df.columns:
+        state_map = dict(zip(states_df['i'], states_df['state']))
+    else:
+        print("Warning: No electronic state column found in states DataFrame")
```

Comment:
You should raise a proper Warning. Not print!
```suggestion
        warnings.warn("Warning: No electronic state column found in states DataFrame")
```